torchmetrics/detection/map.py

```diff
@@ -731,3 +735,74 @@ def compute(self) -> dict:
         metrics.map_per_class = map_per_class_values
         metrics[f"mar_{self.max_detection_thresholds[-1]}_per_class"] = mar_max_dets_per_class_values
         return metrics
+
+
+class MAP(MeanAveragePrecision):
+    r"""
+    Computes the `Mean-Average-Precision (mAP) and Mean-Average-Recall (mAR)
+    <https://jonathan-hui.medium.com/map-mean-average-precision-for-object-detection-45c121a31173>`_
+    for object detection predictions.
+
+    .. deprecated:: v0.7
+        Use :class:`torchmetrics.detect.MeanAveragePrecision`. Will be removed in v0.8.
+
+    Example:
+        >>> import torch
+        >>> preds = [
+        ...   dict(
+        ...     boxes=torch.Tensor([[258.0, 41.0, 606.0, 285.0]]),
+        ...     scores=torch.Tensor([0.536]),
+        ...     labels=torch.IntTensor([0]),
+        ...   )
+        ... ]
+        >>> target = [
+        ...   dict(
+        ...     boxes=torch.Tensor([[214.0, 41.0, 562.0, 285.0]]),
+        ...     labels=torch.IntTensor([0]),
+        ...   )
+        ... ]
+        >>> metric = MAP()  # doctest: +SKIP
```

Comment:
I think import is missing for doctest.

Reply:
I do not want to show a user an import for deprecated function but still want to keep this doctest as validation that it still works as expected... :rabbit: